packages/cli/src/config.ts

```diff
@@ -0,0 +1,35 @@
+import { existsSync } from "fs-extra";
+import { getPackageJSON } from "./paths";
+import { writeFileSync } from "fs";
+
+export const MAILING_CONFIG_FILE = "./mailing.config.json";
+
+export function looksLikeTypescriptProject(): boolean {
+  if (existsSync("./tsconfig.json")) {
+    return true;
+  }
+
+  const pkg = getPackageJSON();
+  return !!(pkg.devDependencies?.typescript || pkg.dependencies?.typescript);
+}
+
+export function writeDefaultConfigFile(): void {
+  const fs = require("fs");
+  const prettier = require("prettier");
+
+  if (!existsSync(MAILING_CONFIG_FILE)) {
+    const configJsonString = prettier.format(
+      JSON.stringify({
```

Comment:
@psugihara it currently is setup to write a default config file with 2 options specified: "typescript" and "emailsDir".  Are there other options that should be set in the default config file that it generates?  e.g. 'quiet' and 'port' could be added.  should it include all options?  in any case i'll add documentation that specifies all the options

Reply:
nice, this looks good. 

`outDir` option would be good to add (used in export-previews)

Reply:
I think we should remove port as an option for now actually. It's hard coded in a few places, so bugged that we allow it. We could fix and re-add later but I don't think it really matters to have it be configurable rn.

Reply:
Can we support `PORT` environment variable?

Reply:
yeah, it's just some work to get it going. In particular, I think next will error if a path like this is not hard-coded, so we'd need to figure out a way around that: https://github.com/sofn-xyz/mailing/blob/main/packages/cli/src/pages/previews/%5BpreviewClass%5D/%5BpreviewFunction%5D.tsx#L16